packages/block-editor/src/hooks/block-style-variation.js

```diff
@@ -59,6 +60,85 @@ function getVariationNameFromClass( className, registeredStyles = [] ) {
 	return null;
 }
 
+/**
+ * This hook is used to generate new block style variation overrides
+ * based on an incoming theme config. If a matching style is found in the config,
+ * a new override is created and returned.
+ *
+ * @param {Object} config A global styles object, containing settings and styles.
+ * @return {Array} An array of new block variation overrides.
+ */
+export function useUpdateBlockStyleVariationOverridesWithConfig( config ) {
```

Comment:
I don't really understand any of this. What about these style overrides makes them revision-specific?